presto-cli/src/main/java/com/facebook/presto/cli/Console.java

```diff
@@ -80,6 +81,26 @@
     @Inject
     public ClientOptions clientOptions = new ClientOptions();
 
+    @SuppressWarnings("ClassExplicitlyExtendsThread")
+    private static void interruptThreadOnExit()
+    {
+        Thread thread = Thread.currentThread();
+        Runtime.getRuntime().addShutdownHook(new Thread()
+        {
+            @Override
+            public void run()
+            {
+                thread.interrupt();
+                try {
+                    thread.join(MAIN_JOIN_TIMEOUT);
```

Comment:
because the thread running the shutdown hook will just call `interrupt()` (and just set a flag) and the vm may exit even before the main thread checks for the interrupted status, which I have seen during testing.